File: clocking-in-clockify/src/slack.ts
import fetch from "node-fetch";

const SLACK_BOT_TOKEN = process.env.SLACK_BOT_TOKEN || "";
if (!SLACK_BOT_TOKEN.length) {
  throw new Error("SLACK_BOT_TOKEN is undefined");
}

export async function post(channel: string, text: string) {
  try {
    const response = await fetch("https://slack.com/api/chat.postMessage", {
      method: "POST",
      headers: {
        Authorization: `Bearer ${SLACK_BOT_TOKEN}`,
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        channel,
        text,
        as_user: true
      })
    });

    const data = await response.json();
    if (!data.ok) {
      throw new Error("Sending failed - " + data.message);
    }

    return data;
  } catch (err) {
    console.error(err);
    throw new Error("Problem");
  }
}

File: Slack.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace SlackApi
{
    public class Slack
    {
        private readonly HttpClient _client;

        public Slack(string authToken)
        {
            _client = new HttpClient();
            _client.DefaultRequestHeaders.Add("Authorization", $"Bearer {authToken}");
        }

        public async Task<PostMessageResponse> PostAsync(string channel, string message)
        {
            var payload = new PostMessagePayload()
            {
                Channel = channel,
                Text = message
            };

            var response = await _client.PostAsync(
                "https://slack.com/api/chat.postMessage",
                new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json")
            );

            response.EnsureSuccessStatusCode();

            var body = await response.Content.ReadAsStringAsync();
            return JsonSerializer.Deserialize<PostMessageResponse>(body);
        }
    }
}
